### PR TITLE
Adds support for snowflake

### PR DESCRIPTION
**Pull request by richiverse**
There are some backwards incompatible changes made to csv module as there were conflicts in the later versions of pandas. I have also split out the extra dependencies for pyarrow, snowflake, and excel in separate modules

---
This change is [Reviewable](https://reviewable.io/reviews/newsela/pandas_ext/26)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -60,7 +60,12 @@ def find_meta(meta):
     version=find_meta("version"),
     license=find_meta("license"),
     install_requires=read("requirements/requirements.in"),
-    extras_require=dict(xls=["xlwt"], xlsx=["openpyxl", "xlsxwriter"]),
+    extras_require=dict(
+        xls=["xlwt"],
+        xlsx=["openpyxl", "xlsxwriter"],
+        snowflake=["setuptools>=41.0.1", "snowflake-sqlalchemy"],
```
Comment (lukeorland) [inline setup.py:66]: Are you sure upgrading setuptools was required?

File: pandas_ext/snowflake.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    return pd.read_sql(sql, con, **kwargs)
+
+
+def to_snowflake(
+        df,
+        schema,
+        table,
+        con=None,
+        index=False,
+        if_exists='append',
```
Comment (lukeorland) [inline pandas_ext/snowflake.py:54]: I think the options for the `if_exists` argument should be documented.

File: pandas_ext/snowflake.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            warehouse=warehouse,
+            role=role,
+            **kwargs
+        )
+    )
+
+
+def read_snowflake(sql: str, con=None, **kwargs) -> pd.DataFrame:
+    """Retrieves a query from snowflake.
+
```
Comment (lukeorland) [inline pandas_ext/snowflake.py]: There is some trailing whitespace in a couple of these docstrings.

**Review by lukeorland — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (richiverse) [reply setup.py:66]: yes, upgrading solved the issue I was having when using snowflake-sqlalchemy

File: pandas_ext/snowflake.py (hunk 2 above)
Comment (richiverse) [reply pandas_ext/snowflake.py:54]: They are documented in the `to_sql` function in pandas. I can post the URL to that page.

**Comment by richiverse** [comment]
@lukeorland all set

**Review by lukeorland — APPROVED** [review]

File: pandas_ext/snowflake.py [hunk 4]
```diff
@@ -65,7 +76,7 @@ def to_snowflake(
 
 
 if __name__ == '__main__':
-    df = pd.DataFrame(dict(a=[1,2,3],b=[4,5,6]))
+    df = pd.DataFrame(dict(a=[1, 2, 3], b=[4, 5, 6]))
```
Comment (lukeorland) [inline pandas_ext/snowflake.py]: Shouldn't this be in a test module, not the `pandas_ext.snowflake` module?

**Pull request merged**